tajo-client/src/main/java/org/apache/tajo/client/SessionConnection.java

```diff
@@ -98,33 +106,49 @@ public SessionConnection(@NotNull ServiceTracker tracker, @Nullable String baseD
     this.manager.setRetries(properties.getInt(RpcConstants.RPC_CLIENT_RETRY_MAX, RpcConstants.DEFAULT_RPC_RETRIES));
     this.userInfo = UserRoleInfo.getCurrentUser();
 
-    try {
-      this.client = getTajoMasterConnection();
-    } catch (ServiceException e) {
-      throw new IOException(e);
-    }
+    this.client = getTajoMasterConnection();
   }
 
   public Map<String, String> getClientSideSessionVars() {
     return Collections.unmodifiableMap(sessionVarsCache);
   }
 
-  public synchronized NettyClientBase getTajoMasterConnection() throws ServiceException {
-    if (client != null && client.isConnected()) return client;
-    else {
+  public synchronized NettyClientBase getTajoMasterConnection() throws SQLException {
+
+    if (client != null && client.isConnected()) {
+      return client;
+    } else {
+
       try {
         RpcClientManager.cleanup(client);
+
         // Client do not closed on idle state for support high available
-        this.client = manager.newClient(getTajoMasterAddr(), TajoMasterClientProtocol.class, false,
-            manager.getRetries(), 0, TimeUnit.SECONDS, false);
+        this.client = manager.newClient(
+            getTajoMasterAddr(),
+            TajoMasterClientProtocol.class,
+            false,
+            manager.getRetries(),
+            0,
+            TimeUnit.SECONDS,
+            false);
         connections.incrementAndGet();
-      } catch (Exception e) {
-        throw new ServiceException(e);
+
+      } catch (Throwable t) {
+        throw SQLExceptionUtil.makeUnableToEstablishConnection(t);
       }
+
       return client;
     }
   }
 
+  protected BlockingInterface getTMStub() throws SQLException {
```

Comment:
As I mentioned above, I think abbreviation is not intuitive because it can be understood in various ways.